Automattic/jetpack
# Adds feature where if review requirement not met, then request

Fixes #

## Proposed changes:

* Adds feature where if a review requirement is not met, then request the review. This is beneficial so that a developer does not need to look at the review requirement(s) to determine who to request reviews from in the case that review requirements are not met.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create `.github/required-review.yaml` similar to the following:
```
- name: Terraform
  paths:
    - '**/*.tf'
  teams:
    - any-of:
      - all-of:
        - team1
        - team2
      - '@user1'

- name: CODEOWNERS
  paths:
    - 'CODEOWNERS'
  teams:
    - team1
    - '@user2'

- name: .github
  paths:
    - '.github/**'
  teams:
    - all-of:
      - team1
      - team2
```
* Make updates to each of the files, as needed
* Observe that on first pass the necessary teams/users are requested

Tested this by deploying fork to my local github env and using settings above. 
* Observed initial failure on all requirements and observed required reviews were requested:
![Screenshot 2023-05-11 at 10 59 30 AM](https://github.com/Automattic/jetpack/assets/52933995/5870f74e-7a2e-46e6-aef0-c0725fc603e7)
* Received review from team1 and observed CODEOWNERS check passed:
![Screenshot 2023-05-11 at 11 02 43 AM](https://github.com/Automattic/jetpack/assets/52933995/a6c197d2-a9b1-4627-8f72-90fead30ccb9)
* Received review from user1 and observed Terraform check passed:
![Screenshot 2023-05-11 at 11 03 51 AM](https://github.com/Automattic/jetpack/assets/52933995/9f5fbcb4-d4b2-4d0b-a85e-5b0a82531c9d)
* Received review from team2 and observed .github check passed:
![Screenshot 2023-05-11 at 11 04 54 AM](https://github.com/Automattic/jetpack/assets/52933995/6e39780a-1709-48eb-a457-8569d040b32c)
* Additionally received review from user2, which had no affect on status.
